src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -772,14 +790,81 @@ protected bool IsAlive
+        }
+
         public Element[] GetJoinedElements()
         {
             return JoinGeometryUtils.GetJoinedElements(Document, this.InternalElement)
                 .Select(id => Document.GetElement(id).ToDSType(true))
                 .ToArray();
         }
 
+        public IEnumerable<Element> SwitchJoinOrder(Element otherElement)
+        {
+            List<Element> joinedElements = new List<Element>();
+            if (JoinGeometryUtils.AreElementsJoined(Document, this.InternalElement, otherElement.InternalElement))
+            {
+                TransactionManager.Instance.EnsureInTransaction(Document);
+                JoinGeometryUtils.SwitchJoinOrder(Document, this.InternalElement, otherElement.InternalElement);
+                TransactionManager.Instance.TransactionTaskDone();
+                joinedElements.AddRange(new List<Element>() { this, otherElement });
+            }
+            return joinedElements;
+        }
+
+        public IEnumerable<Element> GetIntersectingElementsOfCategory(Category category)
+        {
+            BuiltInCategory builtInCategory = (BuiltInCategory)System.Enum.Parse(typeof(BuiltInCategory),
+                                                                                 category.InternalCategory.Id.ToString());
+
+            ElementIntersectsElementFilter filter = new ElementIntersectsElementFilter(this.InternalElement);
+            FilteredElementCollector intersecting = new FilteredElementCollector(Document).WherePasses(filter)
+                                                                                          .OfCategory(builtInCategory);
+            return intersecting.Select(x => x.ToDSType(true)).ToList();
+        }
+
+        /// <summary>
+        /// Unjoins elements if they are joined 
+        /// </summary>
+        /// <param name="elements">List of elements to unjoin</param>
+        /// <returns>Elements that have been unjoined</returns>
+        public static List<Element> UnjoinGeometry(List<Element> elements)
+        {
+            List<Element> modifiedElements = new List<Element>();
+            for (int i = 0; i < elements.Count; i++)
+            {
+                Element[] joinedElements = elements[i].GetJoinedElements();
+                if (joinedElements.Length > 0)
+                {
+                    for (int j = 0; j < joinedElements.Length; j++)
+                    {
+                        JoinGeometryUtils.UnjoinGeometry(
+                            Document,
+                            elements[i].InternalElement,
+                            joinedElements[j].InternalElement);
+
+                        // check if the unjoined element is already in the modifiedElements list 
+                        // if not we add it here
+                        if (!modifiedElements.Any(item => item.Id == joinedElements[j].Id))
+                        {
+                            modifiedElements.Add(joinedElements[j]);
```

Comment:
should we return a list of lists ? that way users can use the output as it's grouped by elements.
if so, we would create an intermediate list for the inner `for` loop and add that to the master list
it would also reduce the number of times we're checking ids are already added to once.